hydra-plutus/src/Hydra/Contract/Commit.hs

```diff
@@ -98,9 +98,12 @@ validator (_party, _headScriptHash, _commit, headId) r ctx =
     ViaAbort ->
       traceIfFalse "ST not burned" (mustBurnST (txInfoMint $ scriptContextTxInfo ctx) headId)
     ViaCollectCom ->
-      traceIfFalse "ST is missing in the output" (hasST headId outputs)
+      traceIfFalse "ST is missing in the output" (hasST headId headOutputValue)
  where
-  outputs = foldMap txOutValue $ txInfoOutputs $ scriptContextTxInfo ctx
+  headOutputValue =
+    case txInfoOutputs (scriptContextTxInfo ctx) of
+      [] -> mempty
+      (headOutput : _otherOutputs) -> txOutValue headOutput
```

Comment:
Could: write `head $ txInfoOutputs $ scriptContextTxInfo ctx` for a nice intended pun :) Also, we don't expect an empty list of outputs and failing is fine in Plutus.

But maybe the error would be puzzling. Just thought that it would be fun to do it that way.

Reply:
And of course the first output here would not even be "the head output".

Reply:
Nevermind.. I got confused as well and this is fine. This could would run on the collect transaction, whose first output is the head output!